### PR TITLE
CB-7-Router

### PR DESCRIPTION
**Pull request by sunnyguan**
* **Please check if the PR fulfills these requirements**
- [✔️] The pull request title contains the ticket number in the title. For example [CB-1]

* **What kind of change does this PR introduce?** (Bug fix, new feature...)

Adds React Router to enable routing between landing page and other pages.

* **What is the current behavior? What does the code do before you made the new changes?** 

Routing is not currently available and only the landing page can be accessed.

* **What is the new behavior?**

Routing now enables the user input page to be accessed at `input/`.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR? Run a new command?)

No, this adds in additional routing functionality that allows for more pages to be shown.

* **Other information**:

Need to replace placeholder component in `App.js` with the actual input component.

### CODE REVIEW

**Review by DHua5922 — APPROVED** [review]

File: frontend/src/App.test.js [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
-      hasBeenRendered(wrapper);
+    hasBeenRendered(wrapper);
   });
 
   it("should render navbar", () => {
-      hasBeenRendered(wrapper.find(MyNavbar));
+    hasBeenRendered(wrapper.find(MyNavbar));
   });
+
+  // issues with enzyme + React 17: https://github.com/enzymejs/enzyme/issues/2462
```
Comment (Codubee) [inline frontend/src/App.test.js:19]: We might have to change to react 16 if we decide to start using mount. Lets discuss this as a team.

File: frontend/src/components/SamplePage/SamplePage.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+function SamplePage() {
+  return (
+    <p>Placeholder for routing</p>
```
Comment (Codubee) [inline frontend/src/components/SamplePage/SamplePage.js:5]: Just wanted to add a note here that we should remove the sample page before we merge this.

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,14 +1,27 @@
 import logo from './logo.svg';
 import './App.css';
 import LandingPage from './components/LandingPage/LandingPage';
+import SamplePage from './components/SamplePage/SamplePage';
```
Comment (Codubee) [inline frontend/src/App.js:4]: Just wanted to add a note here that we should remove the sample page before we merge.

**Review by Codubee — CHANGES_REQUESTED** [review]
Great work Sunny! No major changes to note. Just wanted to discuss moving to react16 for enzyme.

**Comment by sunnyguan** [comment]
Tests have been updated with the latest commit; should I remove the sample page or delete it when the actual input page is merged?

**Comment by Codubee** [comment]
Lets delete it when we are ready to merge.

**Review by DHua5922 — APPROVED** [review]
Test cases looks good. Clicking on user input in navbar goes to sample page.

File: frontend/src/App.js (hunk 3 above)
Comment (DHua5922) [reply frontend/src/App.js:4]: > Great work Sunny! No major changes to note. Just wanted to discuss moving to react16 for enzyme.

Before we downgrade to react 16, try this [unofficial adapter](https://www.npmjs.com/package/@wojtekmaj/enzyme-adapter-react-17) for react 17 for enzyme. This should help with the mount issue.

**Review by Codubee — APPROVED** [review]

**Pull request merged**